Does not work well when using postgresql

In the doc, it says casdoor can create a database if db does not exist by adding an arg `createDatabase` like:

`./casdoor -createDatabase`

If we set dataSourceName = "user=towerhe dbname=casdoor password=towerhe host=localhost port=5432 sslmode=disable", we will got a panic.

```
panic: pq: database "casdoor" does not exist

goroutine 1 [running]:
github.com/casdoor/casdoor/object.InitAdapter()
	/Users/towerhe/Repository/codes/alkaid/alkaid-iam/backend/_server/object/ormer.go:71 +0x1e5
main.main()
	/Users/towerhe/Repository/codes/alkaid/alkaid-iam/backend/_server/main.go:34 +0x46
```

If we remove dbname=casdoor from dataSourceName, the db will be created. But we will got another panic, since we have no dbname in the dataSourceName.

```
panic: dbname is empty

goroutine 1 [running]:
github.com/casdoor/casdoor/object.(*Ormer).open(0xc000b88f40)
	/Users/towerhe/Repository/codes/alkaid/alkaid-iam/backend/_server/object/ormer.go:169 +0xb2
github.com/casdoor/casdoor/object.NewAdapter({0xc0000b0ce0, 0x8}, {0xc0000f22d0, 0x46}, {0xc0000b0d16, 0x3})
	/Users/towerhe/Repository/codes/alkaid/alkaid-iam/backend/_server/object/ormer.go:117 +0xd0
github.com/casdoor/casdoor/object.InitAdapter()
	/Users/towerhe/Repository/codes/alkaid/alkaid-iam/backend/_server/object/ormer.go:75 +0xe8
main.main()
	/Users/towerhe/Repository/codes/alkaid/alkaid-iam/backend/_server/main.go:34 +0x46
```

--- a/object/ormer.go
+++ b/object/ormer.go
@@ -123,12 +123,14 @@ func NewAdapter(driverName string, dataSourceName string, dbName string) *Ormer 
 	return a
 }
 
+func refineDataSourceNameForPostgres(dataSourceName string) string {
+	reg := regexp.MustCompile(`dbname=[^ ]+\s*`)
+	return reg.ReplaceAllString(dataSourceName, "")
+}
+
 func createDatabaseForPostgres(driverName string, dataSourceName string, dbName string) error {
 	if driverName == "postgres" {
-		reg := regexp.MustCompile(`dbname=[^ ]+\s*`)
-		dsn := reg.ReplaceAllString(dataSourceName, "")
-
-		db, err := sql.Open(driverName, dsn)
+		db, err := sql.Open(driverName, refineDataSourceNameForPostgres(dataSourceName))
 		if err != nil {
 			return err
 		}

--- a/object/ormer.go
+++ b/object/ormer.go
@@ -18,6 +18,7 @@ import (
 	"database/sql"
 	"flag"
 	"fmt"
+	"regexp"
 	"runtime"
 	"strings"
 
@@ -124,7 +125,10 @@ func NewAdapter(driverName string, dataSourceName string, dbName string) *Ormer 
 
 func createDatabaseForPostgres(driverName string, dataSourceName string, dbName string) error {
 	if driverName == "postgres" {
-		db, err := sql.Open(driverName, dataSourceName)
+		reg := regexp.MustCompile(`dbname=[^ ]+\s*`)
+		dsn := reg.ReplaceAllString(dataSourceName, "")
+
+		db, err := sql.Open(driverName, dsn)
 		if err != nil {
 			return err
 		}

